pkg/core/blockchain.go

```diff
@@ -805,8 +805,20 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 		bc.lastBatch = cache.DAO.GetBatch()
 	}
 	if bc.config.RemoveUntraceableBlocks {
-		if block.Index > bc.config.MaxTraceableBlocks {
-			index := block.Index - bc.config.MaxTraceableBlocks // is at least 1
+		var start, stop uint32
+		if bc.config.P2PStateExchangeExtensions {
+			// remove batch of old blocks starting from P2-MaxTraceableBlocks-StateSyncInterval up to P2-MaxTraceableBlocks
+			if block.Index > uint32(bc.config.StateSyncInterval)+bc.config.MaxTraceableBlocks && int(block.Index)%bc.config.StateSyncInterval == 0 {
```

Comment:
bc.config.StateSyncInterval*2?

Reply:
Not yet fixed, and I still need to adjust carefully all the places where StateSyncInterval is used.

Reply:
Fixed.